resonance_audio/utils/lockless_task_queue.h

```diff
@@ -51,6 +51,19 @@ class LocklessTaskQueue {
   void Clear();
 
  private:
+  struct tagged_ptr_atomic {
+    std::atomic<uint32_t> offset;
+    std::atomic<uint32_t> tag;
+  };
+
+  // Union for swap operations
+  union NodeAndTag {
+    std::atomic<uint64_t> both_atomic;
```

Comment:
This looks dangerous to me given that the size of the std::atomic is not guaranteed to be the exact size of its templated type, see also discussion here: https://stackoverflow.com/questions/18278611/can-two-stdatomics-be-part-of-one-union